tests/components/zamg/test_init.py

```diff
@@ -0,0 +1,144 @@
+"""Test Zamg component init."""
+from unittest.mock import MagicMock
+
+import pytest
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.components.weather import DOMAIN as WEATHER_DOMAIN
+from homeassistant.components.zamg.const import DOMAIN as ZAMG_DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from . import FIXTURE_CONFIG_ENTRY
+from .conftest import (
+    TEST_STATION_ID,
+    TEST_STATION_ID_2,
+    TEST_STATION_NAME,
+    TEST_STATION_NAME_2,
+)
+
+from tests.common import MockConfigEntry
+
+
+@pytest.mark.parametrize(
+    "entitydata,old_unique_id,new_unique_id",
+    [
+        (
+            {
+                "domain": SENSOR_DOMAIN,
+                "platform": ZAMG_DOMAIN,
+                "unique_id": f"{TEST_STATION_NAME}_{TEST_STATION_ID}",
+                "suggested_object_id": f"Zamg {TEST_STATION_NAME}",
+                "disabled_by": None,
+            },
+            f"{TEST_STATION_NAME}_{TEST_STATION_ID}",
+            TEST_STATION_ID,
+        ),
+        (
+            {
+                "domain": SENSOR_DOMAIN,
+                "platform": ZAMG_DOMAIN,
+                "unique_id": f"{TEST_STATION_NAME_2}_{TEST_STATION_ID_2}",
+                "suggested_object_id": f"Zamg {TEST_STATION_NAME_2}",
+                "disabled_by": None,
+            },
+            f"{TEST_STATION_NAME_2}_{TEST_STATION_ID_2}",
+            TEST_STATION_ID_2,
+        ),
+        (
+            {
+                "domain": WEATHER_DOMAIN,
+                "platform": ZAMG_DOMAIN,
+                "unique_id": f"{TEST_STATION_NAME_2}_{TEST_STATION_ID_2}_temperature",
+                "suggested_object_id": f"Zamg {TEST_STATION_NAME_2}",
+                "disabled_by": None,
+            },
+            f"{TEST_STATION_NAME_2}_{TEST_STATION_ID_2}_temperature",
+            f"{TEST_STATION_ID_2}_temperature",
```

Comment:
The weather domain entity should not have a suffix to the unique_id.

Reply:
This test is just to check if the unique_id of the sensor stays untouched.